### PR TITLE
PyObjectHandle simplification

### PR DESCRIPTION
**Pull request by pepone**
This is a minor simplification for PyObjectHandle implementation.

### CODE REVIEW

File: python/modules/IcePy/Util.cpp [hunk 1]
```diff
@@ -231,30 +231,26 @@ IcePy::PyObjectHandle::PyObjectHandle(PyObject* p) : _p(p) {}
...
 IcePy::PyObjectHandle::PyObjectHandle(const PyObjectHandle& p) : _p(p._p) { Py_XINCREF(_p); }
 
-IcePy::PyObjectHandle::~PyObjectHandle()
-{
-    if (_p)
-    {
-        Py_DECREF(_p);
-    }
-}
+IcePy::PyObjectHandle::~PyObjectHandle() { Py_XDECREF(_p); }
```
Comment (pepone) [inline python/modules/IcePy/Util.cpp:234]: `Py_XDECREF` and `Py_XINCREF` include null handling.

See https://docs.python.org/3/c-api/refcounting.html

File: python/modules/IcePy/Util.cpp [hunk 2]
```diff
@@ -231,30 +231,26 @@ IcePy::PyObjectHandle::PyObjectHandle(PyObject* p) : _p(p) {}
...
 }
 
-void
+IcePy::PyObjectHandle&
 IcePy::PyObjectHandle::operator=(const PyObjectHandle& p)
 {
-    Py_XDECREF(_p);
-    _p = p._p;
-    Py_XINCREF(_p);
+    if (this != &p)
```
Comment (pepone) [inline python/modules/IcePy/Util.cpp:247]: handle self assignment

File: python/modules/IcePy/Util.cpp [hunk 3]
```diff
@@ -231,30 +231,26 @@ IcePy::PyObjectHandle::PyObjectHandle(PyObject* p) : _p(p) {}
...
-{
-    if (_p)
-    {
-        Py_DECREF(_p);
-    }
-}
+IcePy::PyObjectHandle::~PyObjectHandle() { Py_XDECREF(_p); }
 
-void
+IcePy::PyObjectHandle&
```
Comment (pepone) [inline python/modules/IcePy/Util.cpp:236]: Seems uncommon to use `void` for the `operator=` return type.

**Review by bernardnormier — APPROVED** [review]
Looks good.

I am wondering if we should also make the PyObject* ctor explicit.

**Pull request merged**